R/docvars.R

```diff
@@ -239,19 +239,31 @@ set_docvars <- function(dvars, field = NULL, value)  {
     } else {
         dvars_system <- select_fields(dvars, c("text", "system"))
         dvars_user <- select_fields(dvars, "user")
+        
         if (is.dfm(value))
             value <- convert(value, to = "data.frame")[, -1, drop = FALSE]
+        
         if (is.null(field)) {
-            if (is.data.frame(value)) {
-                if (nrow(value) != nrow(dvars))
-                    stop(message_error("docvar_mismatch"))
-                result <- cbind(dvars_system, value)
-            } else {
+            if (is.null(value)) {
                 result <- dvars_system
+            } else {
+                if (is.matrix(value))
+                    value <- as.data.frame(value, stringsAsFactors = FALSE)
+                if (is.data.frame(value)) {
+                    if (nrow(value) != nrow(dvars))
+                        stop(message_error("docvar_mismatch"), call. = FALSE)
+                    if (any(missingnames <- nzchar(names(value), keepNA = TRUE))) {
+                        names(value)[which(is.na(missingnames))] <- 
+                            paste0("V", seq_len(sum(is.na(missingnames))))
+                    }
+                    result <- cbind(dvars_system, value)
```

Comment:
If we add check for column names in line 258, assignment of matrix without names will be also blocked.